### PR TITLE
Chore: remove legacy mdx content collections handler

### PR DESCRIPTION
**Pull request by bholmesdev**
## Changes

- Remove content collections mapper for MDX version 0.17.X and below

## Testing

Make sure nothing breaks

## Docs

changelog should suffice!

### CODE REVIEW

**Review by bluwy — APPROVED** [review]

**Comment by bluwy** [comment]
Actually there seems to be a build failure 👀

File: .changeset/six-keys-cheat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+Remove legacy handling for MDX content collections. Ensure you are using @astrojs/mdx v0.18 and above.
```
Comment (ematipico) [inline .changeset/six-keys-cheat.md:5]:
```suggestion
Remove legacy handling for MDX content collections. Ensure you are using `@astrojs/mdx` v0.18 and above.
```

**Review by ematipico — APPROVED** [review]

File: .changeset/six-keys-cheat.md (hunk 1 above)
Comment (sarah11918) [inline .changeset/six-keys-cheat.md]:
```suggestion
Remove legacy handling for MDX content collections. Ensure you are using `@astrojs/mdx` v0.18 or above.
```
Great catch, @ematipico! Would also point out that you can't use ALL the versions including and above v0.18 at once... :wink:

Comment (sarah11918) [inline .changeset/six-keys-cheat.md]:
```suggestion
Remove legacy handling for MDX content collections. Ensure you are using `@astrojs/mdx` v0.18 or above.
```
Great catch, @ematipico! Would also point out that you can't use ALL the versions including and above v0.18 at once... :wink:

Comment (sarah11918) [reply .changeset/six-keys-cheat.md]: @bholmesdev This comment might have literally come in AS you were updating the code formatting from Ema, so I wanted to make sure you saw this nit of *mine*, too!

**Pull request merged**